enclave/enclave-api/src/test/java/com/quorum/tessera/enclave/encoder/V2PayloadEncoder.java

```diff
@@ -0,0 +1,230 @@
+        affectedContractTxs.putLong(payload.getAffectedContractTransactions().size());
+        for (Map.Entry<TxHash, SecurityHash> entry : payload.getAffectedContractTransactions().entrySet()) {
+            affectedContractTxs.putLong(entry.getKey().getBytes().length);
+            affectedContractTxs.put(entry.getKey().getBytes());
+            affectedContractTxs.putLong(entry.getValue().getData().length);
+            affectedContractTxs.put(entry.getValue().getData());
+        }
+        byte[] executionHash = new byte[0];
+        if (Objects.nonNull(payload.getExecHash()) && payload.getExecHash().length > 0) {
+            executionHash = encodeField(payload.getExecHash());
+        }
+
+        return ByteBuffer.allocate(
+                        senderKey.length
+                                + cipherText.length
+                                + nonce.length
+                                + recipients.length
+                                + recipientNonce.length
+                                + recipientBytes.length
+                                + privacyModeByte.length
+                                + affectedContractsPayloadLength
+                                + executionHash.length)
+                .put(senderKey)
+                .put(cipherText)
+                .put(nonce)
+                .put(recipients)
+                .put(recipientNonce)
+                .put(recipientBytes)
+                .put(privacyModeByte)
+                .put(affectedContractTxs.array())
+                .put(executionHash)
+                .array();
+    }
+
+    public V2EncodedPayload decode(final byte[] input) {
+        final ByteBuffer buffer = ByteBuffer.wrap(input);
+
+        final long senderSize = buffer.getLong();
+        final byte[] senderKey = new byte[Math.toIntExact(senderSize)];
+        buffer.get(senderKey);
+
+        final long cipherTextSize = buffer.getLong();
+        final byte[] cipherText = new byte[Math.toIntExact(cipherTextSize)];
+        buffer.get(cipherText);
+
+        final long nonceSize = buffer.getLong();
+        final byte[] nonce = new byte[Math.toIntExact(nonceSize)];
+        buffer.get(nonce);
+
+        final long numberOfRecipients = buffer.getLong();
+        final List<byte[]> recipientBoxes = new ArrayList<>();
+        for (long i = 0; i < numberOfRecipients; i++) {
+            final long boxSize = buffer.getLong();
+            final byte[] box = new byte[Math.toIntExact(boxSize)];
+            buffer.get(box);
+            recipientBoxes.add(box);
+        }
+
+        final long recipientNonceSize = buffer.getLong();
+        final byte[] recipientNonce = new byte[Math.toIntExact(recipientNonceSize)];
+        buffer.get(recipientNonce);
+
+        // this means there are no recipients in the payload (which we receive when we are a participant)
+        // TODO - not sure this is right
```

Comment:
what action is required to verify this?

Reply:
This is out of the scope for this change and can be addressed in a separate PR. The encoder was copied as-is for testing purpose.